### PR TITLE
fix: Fixed method visibility of ClassInspectionService, JavaConstructService, JavaValidationService,StringUtil,ClassHelper and migrated ClassFinder into ClassHelper

### PR DESCRIPTION
**Pull request by sunilmeddr**
fix: Review visibility of ClassInspectionService methods
fix: Review JavaConstructService methods visibility and remove unused
fix: Review ClassFinder/StringUtil/ClassHelper utility methods and remove unused
fix: Review JavaValidationService methods visibility and remove unused

Fixes: #170
Fixes: #184
Fixes: #185
Fixes: #186

### CODE REVIEW

File: runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/StringUtil.java [hunk 1]
```diff
@@ -24,7 +24,7 @@ public static boolean isEmpty(String s) {
         return s == null || "".equals(s.trim());
```
Comment (igarashitm) [inline runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/StringUtil.java]: For me this is also a Java specific utilities and could be migrated into ClassHelper, WDYT @sunilmeddr ?

File: runtime/modules/java/core/src/test/java/io/atlasmap/java/inspect/ClassInspectionServiceTest.java [hunk 2]
```diff
@@ -76,37 +75,6 @@ public void testDetectArrayClass() {
         assertEquals(Map.class, classInspectionService.detectArrayClass(Map[][][][].class));
     }
 
-    @Test
```
Comment (igarashitm) [inline runtime/modules/java/core/src/test/java/io/atlasmap/java/inspect/ClassInspectionServiceTest.java]: I think these test cases should be covered from somewhere else. Same tests should be run via public methods.

File: runtime/modules/java/module/src/main/java/io/atlasmap/java/module/JavaValidationService.java [hunk 3]
```diff
@@ -86,11 +86,6 @@ public void init() {
         versionMap.put("1.1", 45);
     }
 
-    public void destroy() {
```
Comment (igarashitm) [inline runtime/modules/java/module/src/main/java/io/atlasmap/java/module/JavaValidationService.java]: why removing destroy()?

File: runtime/modules/java/module/src/main/java/io/atlasmap/java/module/JavaValidationService.java [hunk 4]
```diff
@@ -106,14 +101,15 @@ protected String getModuleFieldName(JavaField field) {
...
     }
 
-    protected void validateSourceAndTargetTypes(String mappingId, Field inputField, Field outField, List<Validation> validations) {
+    protected void validateSourceAndTargetTypes(String mappingId, Field inputField, Field outField,
+            List<Validation> validations) {
         if ((inputField instanceof JavaField && outField instanceof JavaField)
                 && ((inputField.getFieldType() == null || outField.getFieldType() == null)
-                    || (inputField.getFieldType().compareTo(FieldType.COMPLEX) == 0
-                    || outField.getFieldType().compareTo(FieldType.COMPLEX) == 0))) {
+                        || (inputField.getFieldType().compareTo(FieldType.COMPLEX) == 0
```
Comment (igarashitm) [inline runtime/modules/java/module/src/main/java/io/atlasmap/java/module/JavaValidationService.java]: I know it's not your code, but doesn't it look weird to use `compareTo()` here? I would use just `==` for the Enum

File: runtime/modules/java/module/src/main/java/io/atlasmap/java/module/JavaValidationService.java [hunk 5]
```diff
@@ -201,7 +199,7 @@ private void validateClass(String mappingId, JavaField field, List<Validation> v
         }
     }
 
-    protected Integer detectClassVersion(String className) {
+    Integer detectClassVersion(String className) {
```
Comment (igarashitm) [inline runtime/modules/java/module/src/main/java/io/atlasmap/java/module/JavaValidationService.java]: should it be package private? for what purpose? It just looks like a private routine though.

File: runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/ClassInspectionService.java [hunk 6]
```diff
@@ -777,15 +699,15 @@ protected Integer detectArrayDimensions(Class<?> clazz) {
...
+    private Class<?> detectListClass(ClassLoader classLoader, Field field) throws ClassNotFoundException {
         List<String> types = detectParameterizedTypes(field, true);
         if (types != null && !types.isEmpty()) {
             return classLoader.loadClass(types.get(0));
         }
         return null;
     }
 
-    protected Class<?> detectArrayClass(Class<?> clazz) {
+    Class<?> detectArrayClass(Class<?> clazz) {
```
Comment (igarashitm) [inline runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/ClassInspectionService.java]: why package private?

File: runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/ClassInspectionService.java [hunk 7]
```diff
@@ -55,143 +52,71 @@
...
+    private Boolean getDisablePackagePrivateOnlyFields() {
         return disablePackagePrivateOnlyFields;
     }
 
-    public void setDisablePackagePrivateOnlyFields(Boolean disablePackagePrivateOnlyFields) {
-        this.disablePackagePrivateOnlyFields = disablePackagePrivateOnlyFields;
-    }
-
-    public Boolean getDisablePrivateOnlyFields() {
+    private Boolean getDisablePrivateOnlyFields() {
```
Comment (igarashitm) [inline runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/ClassInspectionService.java]: getter/setter should be public

File: runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/ClassInspectionService.java [hunk 8]
```diff
@@ -55,143 +52,71 @@
...
+    private Boolean getDisableProtectedOnlyFields() {
         return disableProtectedOnlyFields;
     }
 
     public void setDisableProtectedOnlyFields(Boolean disableProtectedOnlyFields) {
         this.disableProtectedOnlyFields = disableProtectedOnlyFields;
     }
 
-    public Boolean getDisablePackagePrivateOnlyFields() {
+    private Boolean getDisablePackagePrivateOnlyFields() {
```
Comment (igarashitm) [inline runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/ClassInspectionService.java]: getter/setter should be public

File: runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/ClassInspectionService.java [hunk 9]
```diff
@@ -55,143 +52,71 @@
...
-    public List<String> getClassNameBlacklist() {
-        return this.classNameBlacklist;
-    }
-
     public List<String> getFieldBlacklist() {
         return this.fieldBlacklist;
     }
 
-    public Boolean getDisableProtectedOnlyFields() {
+    private Boolean getDisableProtectedOnlyFields() {
```
Comment (igarashitm) [inline runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/ClassInspectionService.java]: getter/setter should be public

File: runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/ClassInspectionService.java [hunk 10]
```diff
@@ -55,143 +52,71 @@
...
     private Boolean disablePrivateOnlyFields = false;
     private Boolean disablePublicOnlyFields = false;
     private Boolean disablePublicGetterSetterFields = false;
 
-    public List<String> getListClasses() {
+    private List<String> getListClasses() {
         return this.listClasses;
     }
 
-    public List<String> getClassNameBlacklist() {
```
Comment (igarashitm) [inline runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/ClassInspectionService.java]: I don't think we have to remove getters even if it's not used today...

File: runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/ClassInspectionService.java [hunk 11]
```diff
@@ -55,143 +52,71 @@
...
     private List<String> fieldBlacklist = new ArrayList<String>(Arrays.asList("serialVersionUID"));
-    private List<String> classNameBlacklist = new ArrayList<String>();
     private Boolean disablePackagePrivateOnlyFields = false;
     private Boolean disableProtectedOnlyFields = false;
     private Boolean disablePrivateOnlyFields = false;
     private Boolean disablePublicOnlyFields = false;
     private Boolean disablePublicGetterSetterFields = false;
 
-    public List<String> getListClasses() {
+    private List<String> getListClasses() {
```
Comment (igarashitm) [inline runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/ClassInspectionService.java]: getter/setter should be public

**Comment by igarashitm** [comment]
And there is a conflict, please rebase on top of upstream master

File: runtime/modules/java/module/src/main/java/io/atlasmap/java/module/JavaValidationService.java (hunk 3 above)
Comment (sunilmeddr) [reply runtime/modules/java/module/src/main/java/io/atlasmap/java/module/JavaValidationService.java]: It has no references

Comment (igarashitm) [reply runtime/modules/java/module/src/main/java/io/atlasmap/java/module/JavaValidationService.java]: I would think this kind of lifecycle methods are added for future use...

File: runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/ClassInspectionService.java (hunk 6 above)
Comment (sunilmeddr) [reply runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/ClassInspectionService.java]: out this class it is referenced only in the testcases. Shall I use the protected in these kind of cases

File: runtime/modules/java/core/src/test/java/io/atlasmap/java/inspect/ClassInspectionServiceTest.java (hunk 2 above)
Comment (sunilmeddr) [reply runtime/modules/java/core/src/test/java/io/atlasmap/java/inspect/ClassInspectionServiceTest.java]: Can you please elaborate

File: runtime/modules/java/module/src/main/java/io/atlasmap/java/module/JavaValidationService.java (hunk 5 above)
Comment (sunilmeddr) [reply runtime/modules/java/module/src/main/java/io/atlasmap/java/module/JavaValidationService.java]: out this class this method  is referenced only in the testcases. Shall I continue using the protected

File: runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/ClassInspectionService.java (hunk 10 above)
Comment (sunilmeddr) [reply runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/ClassInspectionService.java]: Class variable classNameBlacklist was removed as it has no references. So getClassNameBlacklist also removed.

Comment (sunilmeddr) [reply runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/ClassInspectionService.java]: Class variable classNameBlacklist was removed as it has no references,so corresponding getxxx method also removed

File: runtime/modules/java/module/src/main/java/io/atlasmap/java/module/JavaValidationService.java (hunk 5 above)
Comment (igarashitm) [reply runtime/modules/java/module/src/main/java/io/atlasmap/java/module/JavaValidationService.java]: Then this should be private and test should be modified to run from public methods... that's the primary goal for this issues...

File: runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/StringUtil.java (hunk 1 above)
Comment (sunilmeddr) [reply runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/StringUtil.java]: All the string based utility methods are in the classStringUtil.java . This is one of the utility method.

File: runtime/modules/java/module/src/main/java/io/atlasmap/java/module/JavaValidationService.java (hunk 3 above)
Comment (sunilmeddr) [reply runtime/modules/java/module/src/main/java/io/atlasmap/java/module/JavaValidationService.java]: Its not look like a framework method as this method is not in the parent class BaseModuleValidationService.java. I will add it back to make sure we dont get any issues

**Comment by igarashitm** [comment]
Also, can you describe about the commit **in the first line**, instead of `This commit contain multiple fixes` and having following lines, like this
```
fix: Fixed method visibility of ClassInspectionService, JavaConstructService, JavaValidationService and migrated all utilities into ClassHelper

Fixes: #...
```

File: runtime/modules/java/module/src/main/java/io/atlasmap/java/module/JavaValidationService.java (hunk 5 above)
Comment (sunilmeddr) [reply runtime/modules/java/module/src/main/java/io/atlasmap/java/module/JavaValidationService.java]: Then I think,in this case we can remove the testcase

**Comment by sunilmeddr** [comment]
Commits from another PR updated the same file,that this pull request tried to merge. I have resolved the issue,by merging the 2 files.

**Comment by sunilmeddr** [comment]
Can somebody review the changes

**Comment by sunilmeddr** [comment]
Can anyone review the changes

File: runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/ClassHelper.java [hunk 12]
```diff
@@ -26,22 +28,26 @@
...
         opts.add(isMethodNameFromFieldName(fieldName));
         return opts;
     }
 
-    public static String setMethodNameFromFieldName(String fieldName) {
-        return "set" + StringUtil.capitalizeFirstLetter(fieldName);
-    }
-
-    public static String getMethodNameFromFieldName(String fieldName) {
+    private static String getMethodNameFromFieldName(String fieldName) {
```
Comment (rohanmars) [inline runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/ClassHelper.java]: why are these private static? seems strange.

File: runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/ClassInspectionService.java (hunk 11 above)
Comment (rohanmars) [reply runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/ClassInspectionService.java]: agreed, should be public

File: runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/StringUtil.java [hunk 13]
```diff
@@ -24,7 +24,7 @@ public static boolean isEmpty(String s) {
...
     }
 
-    public static String capitalizeFirstLetter(String sentence) {
+    static String capitalizeFirstLetter(String sentence) {
```
Comment (rohanmars) [inline runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/StringUtil.java]: why the removal of public keyword for these methods?

**Comment by sunilmeddr** [comment]
@rohanmars , you may be looking at the wrong commit. These issues are already fixed.

https://github.com/atlasmap/atlasmap/pull/201/commits/3e8641a10d5e57de9bfe8cf16faf9a1858bd2d34

**Comment by sunilmeddr** [comment]
I am not able to squash, as in between the commits, I merged the code from
the master due the conflicts( Some other pull request modified the same
file)

**Comment by rohanmars** [comment]
okay. In the future do a rebase from master to your branch instead. This way it will keep your commits "on top" and any merge conflicts will be in a new commit (on top) which you can then squash into your work.

**Comment by rohanmars** [comment]
LGTM :+1:

File: runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/ClassInspectionService.java [hunk 14]
```diff
@@ -66,7 +63,6 @@
             "java.util.concurrent.ConcurrentHashMap", "java.util.concurrent.ConcurrentMap"));
     private AtlasConversionService atlasConversionService = null;
     private List<String> fieldBlacklist = new ArrayList<String>(Arrays.asList("serialVersionUID"));
-    private List<String> classNameBlacklist = new ArrayList<String>();
```
Comment (igarashitm) [inline runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/ClassInspectionService.java]: I thought we agreed several times with @mattrpav that the class name black list is a place holder for the future use and we shouldn't remove it.

File: runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/ClassInspectionService.java [hunk 15]
```diff
@@ -101,10 +93,6 @@ public Boolean getDisablePackagePrivateOnlyFields() {
         return disablePackagePrivateOnlyFields;
     }
 
-    public void setDisablePackagePrivateOnlyFields(Boolean disablePackagePrivateOnlyFields) {
```
Comment (igarashitm) [inline runtime/modules/java/core/src/main/java/io/atlasmap/java/inspect/ClassInspectionService.java]: setter

File: runtime/modules/java/core/src/test/java/io/atlasmap/java/inspect/ClassInspectionServiceTest.java [hunk 16]
```diff
@@ -51,62 +46,6 @@ public void tearDown() {
         classInspectionService = null;
     }
 
-    @Test
```
Comment (igarashitm) [inline runtime/modules/java/core/src/test/java/io/atlasmap/java/inspect/ClassInspectionServiceTest.java]: Yet again, these test cases needs to be tested via `inspectClass()`

**Comment by mattrpav** [comment]
@igarashitm yes, I asked Rohan to make a first pass review as my schedule did not allow this week as originally planned. I’m working with Sunil to update the PR to include those changes.

**Comment by mattrpav** [comment]
@igarashitm those two changes have been incorporated

**Pull request merged**